scripts/install-calico-windows.ps1

```diff
@@ -124,12 +137,12 @@ function GetCalicoKubeConfig()
         Import-Module $eksAWSToolsModulePath
     }
 
-    $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get secret -n kube-system | findstr $SecretName | % { $_.Split(" ") | select -first 1 }
+    $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get secret -n $CalicoNamespace --field-selector=type=kubernetes.io/service-account-token --no-headers -o custom-columns=":metadata.name" | findstr $SecretName | select -first 1
```

Comment:
I don't understand this part. Could you elaborate a bit?

Reply:
`--no-headers -o custom-columns=":metadata.name"` is just a refinement of what this line did before. This just makes kubectl return only the secret names to run `findstr` against.

The field-selector is important. In OCP, I noticed that a secret of type dockerconfig was named `calico-node-dockerXXX` and it would come before the calico-node service-account tokens